components/Auction-Schedule.jsx

```diff
@@ -0,0 +1,157 @@
+
+// Generated at run-time
+let chain = undefined;
+let wsProvider = undefined;
+let api = undefined;
+let auctionSelections = [];
+
+// Component that retrieves and displays on-chain auction data
+function AuctionSchedule() {
+	const [auctions, setAuctions] = useState("Loading Auctions...");
+
+	useEffect(async () => {
+		const title = document.title;
+		// Polkadot
+		if (title === "Parachain Slot Auctions · Polkadot Wiki") {
+			chain = "polkadot"
+			wsProvider = new WsProvider("wss://rpc.polkadot.io");
+			await Connect(wsProvider, firstAuctionBlockDot, setAuctions);
+		} 
+		// Kusama
+		else if (title === "Parachain Slot Auctions · Guide") {
+			chain = "kusama";
+			wsProvider = new WsProvider("wss://kusama-rpc.polkadot.io/");
+			await Connect(wsProvider, firstAuctionBlockKsm, setAuctions);
+		}
+	// Other
+		else {
+			console.log("Unknown wiki/guide type");
+		}
+	}, []);
+
+	// Render
+	if (chain !== undefined) {
+		return auctions;
+	} else {
+		return (<div>Loading Auctions...</div>)
+	}
+}
+
+// Connect to a chain, retrieve required values, generate UI output values
+async function Connect(wsProvider,  initialBlock, setAuctions) {
+	api = await ApiPromise.create({ provider: wsProvider });
+
+	// Get the current block for projection
+	const currentBlock = await api.rpc.chain.getBlock();
+	const currentBlockNumber = currentBlock.block.header.number.toPrimitive();
+
+	currentBlock.block.extrinsics.forEach((ex, index) => {
+		console.log(index, ex.hash.toHex());
+	});
+
+	// Get current on-chain date/time
+	let chainTimestamp = await api.query.timestamp.now();
+	let date = new Date(chainTimestamp.toPrimitive());
+
+	// Add starting block for the given chain
+	let auctions = [];
+	let auctionBlocks = [];
+	auctionBlocks.push(initialBlock);
+
+	// Build auction objects with all required values for UI
+	for (let i = 0; i < auctionCount; i++) {
+		let auction = {};
+
+		// Blocks
+		auction.startBlock = auctionBlocks[auctionBlocks.length - 1];
+		auction.startHash = await api.rpc.chain.getBlockHash(auction.startBlock);
+		const apiAt = await api.at(auction.startHash.toString());
+		const endPeriod = apiAt.consts.auctions.endingPeriod.toPrimitive();
+		const [lease, end] = (await apiAt.query.auctions.auctionInfo()).toJSON();
+		auction.weeksLeased = lease;
+		auction.endPeriodBlock = end;
+		auction.biddingEndsBlock = auction.endPeriodBlock + endPeriod;
+
+		// TODO - estimates should only be made for future block, otherwise use on-chain timestamp
+		auction.startDate = EstimateBlockDate(date, currentBlockNumber, auction.startBlock);
+		auction.endPeriodDate = EstimateBlockDate(date, currentBlockNumber, auction.endPeriodBlock);
+		auction.biddingEndsDate = EstimateBlockDate(date, currentBlockNumber, auction.biddingEndsBlock);
+
+		// TODO - how to get his value?
+		auction.startOnBoard = "December 17th, 2021";
+		auction.endOnBoard = "October 20th, 2023";
+		//auction.endOnBoard = auction.weeksLeased * 7 + auction.startOnBoard;
```

Comment:
Hey @DrW3RK, I made some good progress on this and it renders exactly like the original hard-coded version however I am still a little confused on the timing of some auction events.

I am mainly confused regarding what values from the original table will always be hard-coded vs what can be calculated on-chain.  I made several assumptions that I know are incorrect, but want to pause and confirm which values can't be determined on-chain.  For example, I don't know what happens if not auctions are scheduled or how I could query that information on-chain.

I am also concerned that to generate all the auction estimates for say 25-50 auctions will be slow.  I definitely can optimize from what I am currently doing once I verify the logic but I am pretty sure it will still require at least a couple RPC invocations per auction (maybe can be done in parallel).

I am thinking we take a small step back and load the hard-coded values from json.  This will be 1:1 with the current tool but will now live on the wiki (step 1).  I will follow up with replacing all the possible on-chain values once I have a better understanding of different edge-cases (step 2).

Let me know what you think and I will continue working it out tomorrow.

Reply:
Having a hardcoded auctions schedule on the Wiki for a start sounds good to me. You can implement step 1 in a PR (this one?) and then work on step2 later

Reply:
Yes exactly, I can get the hard-coded version prepared today.  Then I will outline the couple questions I have for our 1:1 or different meeting to finish part 2.  I think there may always be a couple hard-coded values so it will probably just involve removing a few properties from the json schema and pulling them on-chain async.